src/main/battlecode/common/Direction.java

```diff
@@ -277,4 +280,13 @@ private float reduce(float rads) {
         }
         return rads;
     }
+
+    // Stop NaN or infinity directions from messing things up
+    private void assertValid(float num) {
+        if(num == Float.NaN) {
```

Comment:
All comparisons involving NaN (except !=) return false, so even Float.NaN == Float.NaN returns false. This should be Float.isNaN(num) instead. There's also a Float.isInfinite(num) method which might be better to use for the next check.